Firestore/core/test/unit/local/query_engine_test.cc

```diff
@@ -495,13 +497,128 @@ TEST_P(QueryEngineTest, DoesNotIncludeDocumentsDeletedByMutation) {
     EXPECT_TRUE(result.find(kMatchingDocA.key()) != result.end());
     EXPECT_EQ(result.get(kMatchingDocA.key()), kMatchingDocA);
   });
 }
 
-// TODO(orquery): Port test canPerformOrQueriesUsingFullCollectionScan
+TEST_P(QueryEngineTest, CanPerformOrQueriesUsingFullCollectionScan1) {
+  persistence_->Run("CanPerformOrQueriesUsingFullCollectionScan1", [&] {
+    mutation_queue_->Start();
+    index_manager_->Start();
+
+    MutableDocument doc1 = Doc("coll/1", 1, Map("a", 1, "b", 0));
+    MutableDocument doc2 = Doc("coll/2", 1, Map("a", 2, "b", 1));
+    MutableDocument doc3 = Doc("coll/3", 1, Map("a", 3, "b", 2));
+    MutableDocument doc4 = Doc("coll/4", 1, Map("a", 1, "b", 3));
+    MutableDocument doc5 = Doc("coll/5", 1, Map("a", 1, "b", 1));
+    AddDocuments({doc1, doc2, doc3, doc4, doc5});
+
+    // Two equalities: a==1 || b==1.
+    core::Query query1 = Query("coll").AddingFilter(
+        OrFilters({Filter("a", "==", 1), Filter("b", "==", 1)}));
+    DocumentSet result1 = ExpectFullCollectionScan<DocumentSet>(
+        [&] { return RunQuery(query1, kMissingLastLimboFreeSnapshot); });
+    EXPECT_EQ(result1, DocSet(query1.Comparator(), {doc1, doc2, doc4, doc5}));
+
+    // with one inequality: a>2 || b==1.
+    core::Query query2 = Query("coll").AddingFilter(
+        OrFilters({Filter("a", ">", 2), Filter("b", "==", 1)}));
+    DocumentSet result2 = ExpectFullCollectionScan<DocumentSet>(
+        [&] { return RunQuery(query2, kMissingLastLimboFreeSnapshot); });
+    EXPECT_EQ(result2, DocSet(query2.Comparator(), {doc2, doc3, doc5}));
+
+    // (a==1 && b==0) || (a==3 && b==2)
+    core::Query query3 = Query("coll").AddingFilter(
+        OrFilters({AndFilters({Filter("a", "==", 1), Filter("b", "==", 0)}),
+                   AndFilters({Filter("a", "==", 3), Filter("b", "==", 2)})}));
+    DocumentSet result3 = ExpectFullCollectionScan<DocumentSet>(
+        [&] { return RunQuery(query3, kMissingLastLimboFreeSnapshot); });
+    EXPECT_EQ(result3, DocSet(query3.Comparator(), {doc1, doc3}));
+
+    // a==1 && (b==0 || b==3).
+    core::Query query4 = Query("coll").AddingFilter(
+        AndFilters({Filter("a", "==", 1),
+                    OrFilters({Filter("b", "==", 0), Filter("b", "==", 3)})}));
+    DocumentSet result4 = ExpectFullCollectionScan<DocumentSet>(
+        [&] { return RunQuery(query4, kMissingLastLimboFreeSnapshot); });
+    EXPECT_EQ(result4, DocSet(query4.Comparator(), {doc1, doc4}));
+
+    // (a==2 || b==2) && (a==3 || b==3)
+    core::Query query5 = Query("coll").AddingFilter(
+        AndFilters({OrFilters({Filter("a", "==", 2), Filter("b", "==", 2)}),
+                    OrFilters({Filter("a", "==", 3), Filter("b", "==", 3)})}));
+    DocumentSet result5 = ExpectFullCollectionScan<DocumentSet>(
+        [&] { return RunQuery(query5, kMissingLastLimboFreeSnapshot); });
+    EXPECT_EQ(result5, DocSet(query5.Comparator(), {doc3}));
+  });
+}
+
+TEST_P(QueryEngineTest, CanPerformOrQueriesUsingFullCollectionScan2) {
```

Comment:
Because the clang format in Clion complains about having too many checks inside one test case :(

Reply:
Interesting! Ideally each test should really check one thing. But because of all the document setup that would be required, I put them all in one test. Theoretically we could put all the setup code in a different function, and have each test call that function first. Not necessary IMO.